cli/factory.go

```diff
@@ -206,7 +206,12 @@ var initFactoryCommand = cli.Command{
 
 		if runtimeConfig.FactoryConfig.Template {
 			kataLog.WithField("factory", factoryConfig).Info("create vm factory")
-			_, err := vf.NewFactory(ctx, factoryConfig, false)
+			_, err := vf.NewFactory(ctx, factoryConfig, true)
+			if err == nil {
+				err = fmt.Errorf("vm factory is on")
+			} else {
+				_, err = vf.NewFactory(ctx, factoryConfig, false)
```

Comment:
Calling `NewFunction()` twice looks odd - can't we just rely on your new test rather than doing this?

- https://github.com/kata-containers/tests/pull/1481/files

Reply:
Agree, why not expose the init functionality for vm-factory  as a separate function and call that instead.

Reply:
Pushed a new commit that move check to template.go:New.